### ISSUE TITLE
File drag and drop does not work on macOS

### ISSUE THREAD

**Issue by vit9696** [corpnewt/ProperTree#4]
Taking a file and dragging it to ProperTree window in macOS is supposed to load the file in the program. However, it actually does not do it and the GUI remains empty.

**Comment by corpnewt** [comment]
I've looked briefly into this, and it doesn't appear there are any tkinter capabilities for file drag and drop into widgets.  There is an older extension called [TkDND](https://sourceforge.net/projects/tkdnd/) that may be of help here, though.  As this is more a convenience issue - I may forego it until the rest of the script is vetted properly first.

-CorpNewt

**Comment by vit9696** [comment]
Brief googling shows that it may be possible with some tricks: https://stackoverflow.com/a/44889311. I am not positive whether it works on macOS, but I hope it does.

**Comment by corpnewt** [comment]
That appears to be in reference to dragging and dropping only within the tkinter app - and [links to this](https://stackoverflow.com/questions/16143460/drag-and-drop-in-tkinter) with reference to dragging and dropping between applications (Finder -> ProperTree), which leverages the previously mentioned TkDND extension.  

I'll look into it more and see what I can come up with.

-CorpNewt

**Comment by vit9696** [comment]
Right, it indeed does not strictly fit. Yet I spent some time exploring the documentation and it seems that `::tk::mac::OpenDocument` does mention drag and drop support [1]. While it does seem that the issue is not solvable with pure tk, there at least exist some workarounds, which will not require bundling TkDND as it seems [2].

[1] https://www.tcl.tk/man/tcl/TkCmd/tk_mac.htm#M8
[2] https://moosystems.com/articles/8-double-click-on-files-in-finder-to-open-them-in-your-python-and-tk-application.html

**Comment by corpnewt** [comment]
Thank you for the links, they do look promising.  I'll see what I can put together.

-CorpNewt
